source/dsl/GitDSL.ts

```diff
@@ -64,8 +76,7 @@ export interface GitDSL {
    * @param {string} filename the path to the json file
    * @param {string[]} diffTypes An array containing: "add", "del" or "normal" - then you can only get these changes
    */
-
-  diffForFile(filename: string, diffTypes?: string[]): string | null,
+  diffForFile(filename: string, diffTypes?: string[]): Promise<TextDiff | null>,
```

Comment:
Ah, yes that does make sense. @alex3165 does this satisfy your use-case?

i.e. instead of calling `diffForFile('foo.js', ['add'])`, you'd call `diffForFile('foo.js').added` which would return the same sort of string.